lib/log.c

```diff
@@ -65,6 +65,103 @@ const char *zlog_priority[] = {
 	"notifications", "informational", "debugging", NULL,
 };
 
+static char zlog_filters[ZLOG_FILTERS_MAX][ZLOG_FILTER_LENGTH_MAX + 1];
+static uint8_t zlog_filter_count;
+
+static int zlog_filter_lookup(const char *lookup)
+{
+	/* look for a match on the filter in the current filters */
+	for (int i = 0; i < zlog_filter_count; i++) {
+		if (strncmp(lookup, zlog_filters[i], sizeof(zlog_filters[0]))
+		    == 0)
+			return i;
+	}
+	return -1;
+}
+
+void zlog_filter_clear(void)
+{
+	pthread_mutex_lock(&loglock);
+	zlog_filter_count = 0;
+	pthread_mutex_unlock(&loglock);
+}
+
+int zlog_filter_add(const char *filter)
+{
+	pthread_mutex_lock(&loglock);
+
+	if (zlog_filter_count >= ZLOG_FILTERS_MAX) {
+		pthread_mutex_unlock(&loglock);
```

Comment:
I would suggest having `int ret` that you set and then a goto to the unlock at the end of the function instead of having an unlock along multiple return paths